### PR TITLE
fix(VSelect): propagate name attribute to hidden input

### PR DESCRIPTION
**Pull request by cyruscollier**
## Description

Resolves Issue #10152, a regression introduced by merging #9598.

This fixes the missing `name` attribute on the inner hidden input element inside `VSelect`. Since `name` is now removed from the inner text input, it needed to be manually added to the hidden input from the `VSelect` attribute list.

## Motivation and Context

For users that rely on the traditional form input name/value structure to validate and/or submit forms, it's essential that all Vuetify field components maintain this underlying behavior.

## How Has This Been Tested?

This PR Includes unit tests on VSelect and VCombobox to confirm the bug and verify the fix. It also completes the related unit test for the VSelect hidden input value by properly testing that changing the selected item updates the input value.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [x] My code follows the code style of this project.
- [x] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

File: packages/vuetify/src/components/VCombobox/__tests__/VCombobox.spec.ts [hunk 1]
```diff
@@ -319,4 +319,17 @@ describe('VCombobox.ts', () => {
 
     expect(wrapper.vm.$attrs.autocomplete).toBe('on')
   })
+
+  it('should pass the name attribute to the inner input element', async () => {
```
Comment (johnleider) [inline packages/vuetify/src/components/VCombobox/__tests__/VCombobox.spec.ts]: The **genHiddenInput** method is never overwritten and this is already tested in `v-select` https://github.com/vuetifyjs/vuetify/pull/10327/files#diff-5b5fcac8ce361d1dfbffa8cdea0bca32R453.

File: packages/vuetify/src/components/VSelect/VSelect.ts [hunk 2]
```diff
@@ -427,7 +427,7 @@ export default baseMixins.extend<options>().extend({
     genHiddenInput (): VNode {
       return this.$createElement('input', {
         domProps: { value: this.lazyValue },
-        attrs: { type: 'hidden' },
+        attrs: { type: 'hidden', name: this.attrs$.name },
```
Comment (johnleider) [inline packages/vuetify/src/components/VSelect/VSelect.ts]: This should be on 2 lines.

File: packages/vuetify/src/components/VSelect/__tests__/VSelect3.spec.ts [hunk 3]
```diff
@@ -440,11 +440,26 @@ describe('VSelect.ts', () => {
...
+    const wrapper = mountFunction({
+      propsData: {
+        items: ['foo'],
+        name: ['bar'],
+      },
+    })
+
+    const inputs = wrapper.findAll('input')
+    const element = inputs.at(1).element
+    expect(element.name).toEqual('bar')
```
Comment (johnleider) [inline packages/vuetify/src/components/VSelect/__tests__/VSelect3.spec.ts]: There should be a blank line between line 332 and 333.

**Comment by cyruscollier** [comment]
Thanks, @johnleider. Edits have been implemented.

**Pull request merged**